baseband/vdif/base.py

```diff
@@ -361,6 +356,40 @@ def __init__(self, fh_raw, subset=(), sample_rate=None, fill_value=0.,
 
         self._frameset = frameset
 
+    @staticmethod
+    def _get_frame_rate(fh, header):
+        """Returns the number of frames per second in a VDIF file.
+
+        Parameters
+        ----------
+        fh : io.BufferedReader
+            Binary file handle.
+        header_template : header class or instance
+            Definition or instance of file format's header class.
+
+        Returns
+        -------
+        framerate : `~astropy.units.Quantity`
+            Frames per second, in Hz.
+
+        Notes
+        -----
+
+        This function defaults to using `VLBIStreamReaderBase._get_frame_rate`.
+        If that leads to an Exception, it attempts to extract the sample rate
+        from the header, and passes the Exception on if this too is impossible.
+        """
+        try:
+            return super(VDIFStreamReader, VDIFStreamReader)._get_frame_rate(
+                fh, header)
+        except Exception:
+            if 'sample_rate' in header._properties:
+                return int(np.round(
+                    (header.sample_rate /
+                     header.samples_per_frame).to_value(u.Hz))) * u.Hz
```

Comment:
This can be `np.round(header.sample_rate / header.samples_per_frame).to(u.Hz))` -- you don't need the `int` (quantity converts to float), and `round` works on data with units.